crates/hir_def/src/generics.rs

```diff
@@ -60,16 +60,26 @@ pub struct GenericParams {
 /// associated type bindings like `Iterator<Item = u32>`.
 #[derive(Clone, PartialEq, Eq, Debug)]
 pub enum WherePredicate {
-    TypeBound { target: WherePredicateTypeTarget, bound: TypeBound },
-    Lifetime { target: LifetimeRef, bound: LifetimeRef },
+    TypeBound {
+        target: WherePredicateTypeTarget,
+        bound: TypeBound,
+    },
+    Lifetime {
+        target: LifetimeRef,
+        bound: LifetimeRef,
+    },
+    ForLifetime {
+        lifetimes: Box<[LifetimeRef]>,
```

Comment:
these should just be `Name`s, I think; they're not references to existing lifetimes, but defining new ones